docs/content/manual/manual.md

```diff
@@ -0,0 +1,5538 @@
+such as [`add`](#add).
+
+## `reduce`
+
+The `reduce` syntax allows you to combine all of the results of
+an expression by accumulating them into a single answer.
+The form is `reduce EXP as $var (INIT; UPDATE)`.
+As an example, we'll pass `[1,2,3]` to this expression:
+
+    reduce .[] as $item (0; . + $item)
+
+For each result that `.[]` produces, `. + $item` is run to
+accumulate a running total, starting from 0 as the input value.
+In this example, `.[]` produces the results `1`, `2`, and `3`,
+so the effect is similar to running something like this:
+
+    0 | 1 as $item | . + $item |
+        2 as $item | . + $item |
+        3 as $item | . + $item
+
+::: Compatibility
+
+When `UPDATE` yields multiple outputs,
+jq only considers the _last_ one for the next iteration, whereas
+jaq considers _all_ of them.
+For example, the filter
+
+    reduce (0, 1) as $x ([]; . + (["a", $x], ["b", $x]))
+
+yields `["b",0,"b",1]` in jq, whereas in jaq, it yields:
+
+~~~
+["a",0,"a",1]
+["a",0,"b",1]
+["b",0,"a",1]
+["b",0,"b",1]
+~~~
+
+:::
+
+::: Examples
+
+~~~
+reduce .[] as $item (0; . + $item)
+[1,2,3,4,5]
+15
+~~~
+
+~~~
+reduce .[] as [$i,$j] (0; . + $i * $j)
+[[1,2],[3,4],[5,6]]
+44
+~~~
+
+~~~
+reduce .[] as {$x,$y} (null; .x += $x | .y += [$y])
+[{"x":"a","y":1},{"x":"b","y":2},{"x":"c","y":3}]
+{"x":"abc","y":[1,2,3]}
+~~~
+
+:::
+
+## `foreach`
+
+The `foreach` syntax is similar to `reduce`, but intended to
+allow the construction of `limit` and reducers that produce
+intermediate results.
+
+The form is `foreach EXP as $var (INIT; UPDATE; EXTRACT)`.
+As an example, we'll pass `[1,2,3]` to this expression:
+
+    foreach .[] as $item (0; . + $item; [$item, . * 2])
+
+Like the `reduce` syntax, `. + $item` is run for each result
+that `.[]` produces, but `[$item, . * 2]` is run for each
+intermediate values. In this example, since the intermediate
+values are `1`, `3`, and `6`, the `foreach` expression produces
+`[1,2]`, `[2,6]`, and `[3,12]`. So the effect is similar
+to running something like this:
+
+    0 | 1 as $item | . + $item | [$item, . * 2],
+        2 as $item | . + $item | [$item, . * 2],
+        3 as $item | . + $item | [$item, . * 2]
+
+When `EXTRACT` is omitted, the identity filter is used.
+That is, it outputs the intermediate values as they are.
+
+
+::: Note
+
+We can also use a pattern at the place of `$var`.
+
+    foreach EXP as PATTERN (INIT; UPDATE; EXTRACT)
+
+When `PATTERN` binds the variables `$x1`, ..., `$xn`,
+then the expression is equivalent to:
+
+    foreach (EXP as PATTERN | {$x1, ..., $xn}) as $x (
+      INIT;
+      $x as {$x1, ..., $xn} | UPDATE;
+      $x as {$x1, ..., $xn} | EXTRACT
+    )
+
+(Here, the name of `$x` must be chosen such that
+`$x` does not shadow any existing variable at this point.)
+
+A similar transformation can be made for `reduce`.
+
+:::
+
+::: Compatibility
+
+Similarly as for `reduce`,
+jq  considers only the _last_ output of `UPDATE` for the next iteration, whereas
```

Comment:
gojq is compatible with jq in this behavior.

Reply:
Done.